### PR TITLE
Oaks speech

### PR DESCRIPTION
**Pull request by PikalaxALT**

### CODE REVIEW

File: include/oaks_speech_internal.h [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+    int genderSelectFrameBlinkTimer;
+    int lastInputWasTouch;
+    int queuedMsgId;
+    u16 oakPicHTranslateState;
+    s16 oakPicHTranslatePos;
+    OakSpeechYesNo *yesnoMenu;
+    int lastChosenGender; // if the player changes their mind
+} OakSpeechData;
+
+int OakSpeech_DeadstrippedFunction1(int a0);
```
Comment (adrienntindall) [inline include/oaks_speech_internal.h:84]: do these deadstripped functions need to be included in the header or can they be static?

File: src/launch_application.c [hunk 2]
```diff
@@ -1075,15 +1075,15 @@ static void SetName(TaskManager *taskman) {
     FieldSystem *fieldSystem = TaskManager_GetFieldSystem(taskman);
     NamingScreenData *data   = TaskManager_GetEnvironment(taskman);
     switch (data->args->unk0) {
```
Comment (adrienntindall) [inline src/launch_application.c:1077]: unk0 -> state

File: src/launch_application.c [hunk 3]
```diff
@@ -1075,15 +1075,15 @@ static void SetName(TaskManager *taskman) {
...
-    case 0:
+    case NAME_SCREEN_PLAYER:
         PlayerProfile *profile = Save_PlayerData_GetProfileAddr(fieldSystem->saveData);
         Save_Profile_PlayerName_Set(profile, data->args->unk1C);
```
Comment (adrienntindall) [inline src/launch_application.c:1080]: unk1C -> name (playerNameInputString?)

Comment (adrienntindall) [reply src/launch_application.c:1080]: nvm it also gets used for pokemon names, should theoretically also be nameInputString but that's already used for the rival so ???

File: src/oaks_speech.c [hunk 4]
```diff
@@ -0,0 +1,2237 @@
...
+    int gfxId;
+
+    if (data->playerPicShrinkAnimDelay != 0) {
+        --data->playerPicShrinkAnimDelay;
+    } else {
+        ++data->playerPicShrinkAnimStep;
+        data->playerPicShrinkAnimDelay = 8;
+    }
+    if (data->playerGender == PLAYER_GENDER_MALE) {
+        int sp28[6];
```
Comment (adrienntindall) [inline src/oaks_speech.c:1460]: name sp28

File: src/oaks_speech.c [hunk 5]
```diff
@@ -0,0 +1,2237 @@
...
+        ARRAY_ASSIGN(sp28, sPlayerPicShrinkGfx_Male);
+        gfxId = sp28[data->playerPicShrinkAnimStep];
+    } else {
+        int sp10[6];
```
Comment (adrienntindall) [inline src/oaks_speech.c:1464]: name sp10

File: src/oaks_speech_obj.c [hunk 6]
```diff
@@ -0,0 +1,187 @@
...
+        };
+
+        sub_0200CF70(data->spriteRenderer, &oamManagerParam, &oamCharTransferParam, 0x20);
+    }
+
+    sub_0200CFF4(data->spriteRenderer, data->spriteGfxHandler, 10);
+    G2dRenderer_SetSubSurfaceCoords(SpriteRenderer_GetG2dRendererPtr(data->spriteRenderer), 0, FX32_CONST(192));
+
+    {
+        // extern const u16 ov53_021E8740[7];
```
Comment (adrienntindall) [inline src/oaks_speech_obj.c:131]: can remove comment

File: src/oaks_speech_obj.c [hunk 7]
```diff
@@ -0,0 +1,187 @@
...
+            .numOBJmain     = 128,
+            .fromAffineMain = 0,
+            .numAffineMain  = 32,
+            .fromOBJsub     = 0,
+            .numOBJsub      = 128,
+            .fromAffineSub  = 0,
+            .numAffineSub   = 32,
+        };
+
+        // extern const OamCharTransferParam ov53_021E8750;
```
Comment (adrienntindall) [inline src/oaks_speech_obj.c:115]: can remove comment

**Review by adrienntindall — CHANGES_REQUESTED** [review]
Minor things but lgtm

File: include/oaks_speech_internal.h (hunk 1 above)
Comment (PikalaxALT) [reply include/oaks_speech_internal.h:84]: making them static will cause them to be deadstripped at compiletime. they need to be linktime available.

File: src/launch_application.c (hunk 2 above)
Comment (PikalaxALT) [reply src/launch_application.c:1077]: not a state machine but good try

File: src/launch_application.c (hunk 3 above)
Comment (PikalaxALT) [reply src/launch_application.c:1080]: out of scope

File: src/oaks_speech.c (hunk 4 above)
Comment (PikalaxALT) [reply src/oaks_speech.c:1460]: this is a fakematch so the name can stay

File: src/oaks_speech.c (hunk 5 above)
Comment (PikalaxALT) [reply src/oaks_speech.c:1464]: this is a fakematch so the name can stay

File: src/oaks_speech_obj.c (hunk 7 above)
Comment (PikalaxALT) [reply src/oaks_speech_obj.c:115]: the data inlining is in part fakematched so the comment can stay

**Review by adrienntindall — APPROVED** [review]

**Pull request merged**